File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/RolePermissionIdFlattenedDto.java
package org.dddml.wms.domain;


public class RolePermissionIdFlattenedDto
{


    private static String[] _propertyNames = new String[]{
            "roleId",
            "permissionId",
        };

    private static String[] _propertyTypes = new String[]{
            "String",
            "String",
        };

    public static String[] getPropertyNames()
    {
        return _propertyNames;
    }

    public static String[] getPropertyTypes()
    {
        return _propertyTypes;
    }
	
    private RolePermissionId value;

    public RolePermissionIdFlattenedDto()
    {
        this(new RolePermissionId());
    }

    public RolePermissionIdFlattenedDto(RolePermissionId value)
    {
        this.value = value;
    }

    public RolePermissionId toRolePermissionId()
    {
        return this.value;
    }



    public String getRoleId()
    {
        return this.value.getRoleId();
    }

    public void setRoleId(String roleId)
    {
        this.value.setRoleId(roleId);
    }


    public String getPermissionId()
    {
        return this.value.getPermissionId();
    }

    public void setPermissionId(String permissionId)
    {
        this.value.setPermissionId(permissionId);
    }


}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/RolePermissionIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\FlattenedDtoInterface;

class RolePermissionIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'roleId' => 'string',
            'permissionId' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var RolePermissionId
     */
    private $value;

    /**
     * @param RolePermissionId $value
     */
    public function __construct(RolePermissionId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new RolePermissionId();
        }
    }

    /**
     * @return RolePermissionId
     */
    public function toRolePermissionId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getRoleId()
    {
        return $this->value->getRoleId();
    }

    /**
     * @param string $roleId
     */
    public function setRoleId($roleId)
    {
        $this->value->setRoleId($roleId);
    }

    /**
     * @return string
     */
    public function getPermissionId()
    {
        return $this->value->getPermissionId();
    }

    /**
     * @param string $permissionId
     */
    public function setPermissionId($permissionId)
    {
        $this->value->setPermissionId($permissionId);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getRoleId(),
            $this->getPermissionId(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return RolePermissionIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setRoleId($pValues[0]);
        $this->setPermissionId($pValues[1]);
        return $this;
    }

}

